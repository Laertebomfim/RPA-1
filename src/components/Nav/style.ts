import { tv } from 'tailwind-variants';

export const style = tv({
  base: ' flex flex-col items-center md:flex-wrap md:flex-row md:justify-end  bg-white',
});
