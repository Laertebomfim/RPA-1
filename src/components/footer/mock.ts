export const mock = {
  background: true,
  text: `
    <p>
        <a href='http://www.test.com'>Pellentesque habitant morbi tristique senectus</a>
    </p>
   `,
};
