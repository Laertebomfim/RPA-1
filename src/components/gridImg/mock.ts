export const mock = {
  background: true,
  TitleText: 'Lorem ipsum dolor',
  descriptionP:
    'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Molestias libero laborum, repudiandae iure consectetur commodi, error soluta modi dignissimos alias.',
  gridImg: [
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
    {
      TitleAlt: 'text',
      srcImg:
        'https://s1.static.brasilescola.uol.com.br/be/conteudo/images/imagem-em-lente-convexa.jpg',
    },
  ],
};
