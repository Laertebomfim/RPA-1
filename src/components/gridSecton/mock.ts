export const mock = {
  background: true,
  TitleText: 'Lorem ipsum dolor',
  descriptionP:
    'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Molestias libero laborum, repudiandae iure consectetur commodi, error soluta modi dignissimos alias.',
  grid: [
    {
      Title: 'text',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Molestias libero laborum, repudiandae iure consectetur commodi, error soluta modi dignissimos alias expedita, ex culpa nemo velit explicabo mollitia ipsam incidunt debitis!',
    },
    {
      Title: 'text',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Molestias libero laborum, repudiandae iure consectetur commodi, error soluta modi dignissimos alias expedita, ex culpa nemo velit explicabo mollitia ipsam incidunt debitis!',
    },
    {
      Title: 'text',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Molestias libero laborum, repudiandae iure consectetur commodi, error soluta modi dignissimos alias expedita, ex culpa nemo velit explicabo mollitia ipsam incidunt debitis!',
    },
  ],
};
