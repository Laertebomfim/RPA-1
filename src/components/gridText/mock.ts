export const mock = {
  TitleText: 'my title',
  text: ['hi '],
  background: true,
};
