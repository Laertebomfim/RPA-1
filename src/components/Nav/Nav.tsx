import { ReactNode } from 'react';
import LinkComponent from '../headLink/Link';
import { style } from './style';

export type linkPros = { children: ReactNode; http: string; newtab?: '_blank' | '_self' }[];
type NavProps = {
  link: linkPros;
};
export default function Nav({ link }: NavProps) {
  return (
    <nav className={style()}>
      {link.map((object, inds) => (
        <LinkComponent key={inds} http={object.http} newtab={object.newtab}>
          {object.children}
        </LinkComponent>
      ))}
    </nav>
  );
}
