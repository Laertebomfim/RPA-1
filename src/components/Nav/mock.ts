import { linkPros } from './Nav';

export const Mock: linkPros = [
  {
    children: 'link-1',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-2',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-3',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-4',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-5',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-6',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-7',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-8',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-9',
    http: 'http://mock.com',
    newtab: '_self',
  },

  {
    children: 'link-10',
    http: 'http://mock.com',
    newtab: '_self',
  },
];
