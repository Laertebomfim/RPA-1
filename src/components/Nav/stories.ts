import type { Meta, StoryObj } from '@storybook/react';
import Nav from './Nav';
import { Mock } from './mock';
const meta: Meta<typeof Nav> = {
  component: Nav,
};

export default meta;
type story = StoryObj<typeof Nav>;

export const Primary: story = {
  args: {
    link: Mock,
  },
};
