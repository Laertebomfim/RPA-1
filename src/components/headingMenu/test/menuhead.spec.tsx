import { fireEvent, render } from '@testing-library/react';
import HeadingMenu from '../Index';
import { Mock } from '@/components/Nav/mock';

const mockData = [Mock[0]];
describe('<HeadingMenu/>', () => {
  it('create an Snapshot', () => {
    const { container } = render(<HeadingMenu information={mockData} img="" link="" text="logo" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('most stat with menu-infor invisible to mobile', () => {
    const { getByLabelText } = render(
      <HeadingMenu information={mockData} img="" link="" text="logo" />,
    );
    expect(getByLabelText(/menu-infor/i)).toHaveClass('hidden opacity-0');
  });
  it('when he for clicking menu-infor most is visible', () => {
    const { getByLabelText } = render(
      <HeadingMenu information={mockData} img="" link="" text="logo" />,
    );
    fireEvent.click(getByLabelText(/menu-mobile/i));

    expect(getByLabelText(/menu-infor/i)).not.toHaveClass('hidden opacity-0');

    fireEvent.click(getByLabelText(/menu-mobile/i));
    expect(getByLabelText(/menu-infor/i)).toHaveClass('hidden opacity-0');
  });
  it('when clicked it hides the menu', () => {
    const { getByRole, getByLabelText } = render(
      <HeadingMenu information={mockData} img="" link="" text="logo" />,
    );
    fireEvent.click(getByLabelText(/menu-mobile/i));
    fireEvent.click(getByRole('article'));
    expect(getByLabelText(/menu-infor/i)).toHaveClass('hidden opacity-0');
  });
});
