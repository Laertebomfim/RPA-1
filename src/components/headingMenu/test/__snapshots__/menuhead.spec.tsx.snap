// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingMenu/> create an Snapshot 1`] = `
<div
  aria-label="menu-mobile"
  class="fixed p-2 rounded-xl bg-slate-600 top-5 right-10 z-10 md:hidden md:opacity-0"
>
  <svg
    aria-label="open-menu"
    class="lucide lucide-align-justify "
    fill="none"
    height="30"
    stroke="white"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="4"
    viewBox="0 0 24 24"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="3"
      x2="21"
      y1="6"
      y2="6"
    />
    <line
      x1="3"
      x2="21"
      y1="12"
      y2="12"
    />
    <line
      x1="3"
      x2="21"
      y1="18"
      y2="18"
    />
  </svg>
</div>
`;
